recipes/occt/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class OcctConan(ConanFile):
+    name = "occt"
+    description = """a software development platform providing services for 3D
+        surface and solid modeling, CAD data exchange, and visualization."""
+    homepage = "https://github.com/Open-Cascade-SAS/OCCT"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-2.1-or-later"
+    topics = ("3D", "modeling")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+    requires = [("tcl/8.6.10"),
+                ("tk/8.6.10"),
+                ("freetype/2.10.2")]
+    _source_subfolder = "src"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.upper() + "-" + \
+            self.version.replace(".", "_")
+        tools.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            "project (OCCT)",
+            '''project (OCCT)
+                include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+                conan_basic_setup()''')
```

Comment:
The original CMakeLists.txt used a lot of ${CMAKE_SOURCE_DIR} instead of ${CMAKE_CURRENT_SOURCE_DIR}, making it difficult to use the cmake_wrapper. Using it will require a lot more replacing.